### PR TITLE
DENG-2492 create new GA4 derived table firefox_whatsnew_summary_v2

### PR DESCRIPTION
**Pull request by kwindau**
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-2705)

### CODE REVIEW

File: sql/moz-fx-data-marketing-prod/ga_derived/firefox_whatsnew_summary_v2/query.sql [hunk 1]
```diff
@@ -0,0 +1,42 @@
+WITH wnp_visits AS (
+  SELECT
+    date,
+    visit_identifier,
+    TRIM(page_path_level1, '/') AS locale,
+    page_level_2 AS version,
+    mozfun.norm.browser_version_info(page_level_2) AS version_info,
+    country,
+    CAST(bounces AS boolean) AS is_bounce
```
Comment (fbertsch) [inline sql/moz-fx-data-marketing-prod/ga_derived/firefox_whatsnew_summary_v2/query.sql:9]: nit: explicit > implicit
```suggestion
    bounces > 0 AS is_bounce
```

File: sql/moz-fx-data-marketing-prod/ga_derived/firefox_whatsnew_summary_v2/query.sql [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  FROM
+    `moz-fx-data-marketing-prod.ga_derived.www_site_hits_v2`
+  WHERE
+    date = @submission_date
+    AND hit_type = 'PAGE'
+    -- Match page paths like "/{locale}/firefox/{version}/whatsnew/..."
+    -- Version regular expression is adapted from https://github.com/mozilla/bedrock/blob/main/bedrock/releasenotes/__init__.py
+    AND page_level_1 = 'firefox'
```
Comment (fbertsch) [inline sql/moz-fx-data-marketing-prod/ga_derived/firefox_whatsnew_summary_v2/query.sql:17]: What is the difference between `page_path_level1` and `page_level_1`?

File: sql/moz-fx-data-marketing-prod/ga_derived/firefox_whatsnew_summary_v2/query.sql [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  date,
+  country,
+  locale,
+  version,
+  version_info.major_version,
+  version_info.minor_version,
+  version_info.patch_revision,
+  version_info.is_major_release,
+  COUNT(DISTINCT visit_identifier) AS visits,
+  COUNT(DISTINCT CASE WHEN is_bounce = TRUE THEN visit_identifier END) AS bounces
```
Comment (fbertsch) [inline sql/moz-fx-data-marketing-prod/ga_derived/firefox_whatsnew_summary_v2/query.sql:31]: optional, could consider just using IF
```suggestion
  COUNT(DISTINCT IF(is_bounce, visit_identifier, NULL)) AS bounces
```

File: sql/moz-fx-data-marketing-prod/ga_derived/firefox_whatsnew_summary_v2/schema.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  type: NUMERIC
+  description: Patch Revision - The patch revision version number
+- mode: NULLABLE
+  name: is_major_release
+  type: BOOLEAN
+  description: Is Major Release - Indicates if this is a new major release (i.e. minor version is 0)
+- mode: NULLABLE
+  name: visits
+  type: INT64
+  description: Visits - The number of unique visits - combo of user_pseudo_id (cookie ID) and Google Analytics session ID
```
Comment (fbertsch) [inline sql/moz-fx-data-marketing-prod/ga_derived/firefox_whatsnew_summary_v2/schema.yaml:37]: Not sure what you mean by `combo` here, are you saying this is counting the visits for this session and this user?

**Review by fbertsch — APPROVED** [review]
r+wc

File: sql/moz-fx-data-marketing-prod/ga_derived/firefox_whatsnew_summary_v2/schema.yaml (hunk 4 above)
Comment (kwindau) [reply sql/moz-fx-data-marketing-prod/ga_derived/firefox_whatsnew_summary_v2/schema.yaml:37]: Basically just mean the unique # of visits, where a unique visit is from the 2 fields user_pseudo_id and ga_session_id

Comment (kwindau) [reply sql/moz-fx-data-marketing-prod/ga_derived/firefox_whatsnew_summary_v2/schema.yaml:37]: I can try to reword it more clearly

File: sql/moz-fx-data-marketing-prod/ga_derived/firefox_whatsnew_summary_v2/query.sql (hunk 2 above)
Comment (kwindau) [reply sql/moz-fx-data-marketing-prod/ga_derived/firefox_whatsnew_summary_v2/query.sql:17]: Basically page path level 1 is like the first offset, and then page level 1 is the second offset (don't ask me why they named it that way, that's how the old tables were LOL)
Page level 1 = SPLIT(REGEXP_REPLACE(page_location, '^https://www.mozilla.org', ''), '/')[
      SAFE_OFFSET(2)
    ] AS page_level_1

Page path level 1 = SPLIT(REGEXP_REPLACE(all_events.page_location, '^https://www.mozilla.org', ''), '/')[
      SAFE_OFFSET(1)
    ]

**Pull request merged**